fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneMailExampleTest.java

```diff
@@ -0,0 +1,244 @@
+                commit(context);
+            }
+        }
+    }
+
+    private void rebuildIndexMetaData(final FDBRecordContext context, final String document, final Index index) {
+        Pair<FDBRecordStore, QueryPlanner> pair = LuceneIndexTestUtils.rebuildIndexMetaData(context, path, document, index, useRewritePlanner);
+        this.recordStore = pair.getLeft();
+        this.planner = pair.getRight();
+    }
+
+    /**
+     * Load the records for emails from the EML files.
+     * @param dataResourceDirectory the directory name to load EML fields from
+     * @param recordCountToLoad count of records to load
+     * @throws Exception
+     */
+    void loadRecords(String dataResourceDirectory, int recordCountToLoad, int offset) throws Exception {
+        final File file = new File(getClass().getClassLoader().getResource(dataResourceDirectory).getFile());
+        final Properties props = System.getProperties();
+        int index = 0;
+        int mailBoxIndex = 0; // this is ignored if Gaussian distribution is used
+        Random random = new Random();
+
+        long totalWriteTime = 0L;
+        // Write mbox records
+
+        List<String> mboxIds = new ArrayList<>();
+        for (int i = 1; i <= (USE_GAUSSIAN_DISTRIBUTION ? MAIL_BOX_COUNT : MAIL_BOX_DISTRIBUTION.length); i++) {
+            mboxIds.add("box" + i);
+        }
+        for (File emlFile : file.listFiles()) {
+            int mboxId = USE_GAUSSIAN_DISTRIBUTION ? (int)(random.nextGaussian() * (MAIL_BOX_COUNT / 6) + (MAIL_BOX_COUNT / 2 + 1)) : mailBoxIndex + 1;
+            mboxId = Math.min(mboxId, MAIL_BOX_COUNT);
+            mboxId = Math.max(mboxId, 1);
+            Session session = Session.getDefaultInstance(props, null);
+            InputStream source = new FileInputStream(emlFile);
+            MimeMessage message = new MimeMessage(session, source);
+            long start = System.currentTimeMillis();
+            try {
+                if (loadRecord(message, mboxIds.get(mboxId - 1), index + offset)) {
+                    index++;
+                }
+                if (index >= MAIL_BOX_DISTRIBUTION[mailBoxIndex]) {
+                    mailBoxIndex++;
+                }
+                if (index >= recordCountToLoad) {
+                    break;
+                }
+            } finally {
+                totalWriteTime += (System.currentTimeMillis() - start);
+                if (index % 100 == 0) {
+                    start = System.currentTimeMillis();
+                    //zone.flushWithoutReset();
+                    totalWriteTime+= (System.currentTimeMillis()-start);
+                    LOGGER.info("Number of records that have been loaded: {}. Time taken: {} ", index, Duration.ofMillis(totalWriteTime));
+                }
+            }
+        }
+        LOGGER.info("Number of records that have been loaded: {}. Time taken: {} ", index, Duration.ofMillis(totalWriteTime));
```

Comment:
The canonical way to do this (a log message with a static message and some key-value pairs) would be with a `KeyValueLogMessage`. I'd also also suggest including the unit of measure in the logs--something like `load_time_micros`